File: Workspace/Service_Manager/src/main/java/fit/iuh/backend/model/Product.java
package fit.iuh.backend.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; // Đổi từ productId thành id

    private String name;
    private String code; // Thêm code từ Service Catalog
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    private Integer quantity; // Đổi từ stock thành quantity
    private Double price;
    private String status; // Thêm status từ Service Catalog và Cart Order
    private Integer gender;
    private String material;
    private Integer goldKarat;
    private String color;
    private String brand;
    private String size; // Thêm size từ Service Catalog
    private Integer viewCount;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "collection_id")
    private Collection collection;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<ProductImage> imageSet;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "product")
    private List<Review> reviews;

    @OneToMany(mappedBy = "product")
    private List<OrderDetail> orderDetails;

    @OneToMany(mappedBy = "product")
    private List<CartItem> cartItems;
    
    @OneToMany(mappedBy = "product")
    private List<ProductFeature> productFeatures; // Thêm quan hệ productFeatures
    
    @OneToMany(mappedBy = "product")
    private List<ProductVariant> productVariants; // Thêm quan hệ productVariants
    
    @OneToMany(mappedBy = "product")
    private List<ProductSalesSummary> salesSummaries; // Thêm quan hệ salesSummaries
}

File: Workspace/Service_Cart_Order/Service_Cart_Order_Backend/src/models/Product.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/Database');

const Product = sequelize.define('Product', {
    id: { // Đổi từ productID thành id để đồng bộ
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
    },
    name: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    code: { // Thêm code để đồng bộ với Catalog Service
        type: DataTypes.STRING,
        allowNull: true,
    },
    description: {
        type: DataTypes.TEXT, // Đổi từ STRING sang TEXT
        allowNull: true,
    },
    quantity: { // Đổi từ stock thành quantity để đồng bộ
        type: DataTypes.INTEGER,
        allowNull: false,
        defaultValue: 0,
    },
    price: {
        type: DataTypes.DECIMAL(15, 2), // Đổi từ DOUBLE sang DECIMAL
        allowNull: false,
    },
    status: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    gender: {
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    material: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    gold_karat: {
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    color: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    brand: {
        type: DataTypes.STRING,
        allowNull: true,
    },
    size: { // Thêm size để đồng bộ
        type: DataTypes.STRING,
        allowNull: true,
    },
    view_count: { // Thêm view_count để đồng bộ
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    category_id: { // Đổi từ category thành category_id
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    collection_id: { // Thêm collection_id để đồng bộ
        type: DataTypes.INTEGER,
        allowNull: true,
    },
    created_at: {
        type: DataTypes.DATE,
        allowNull: false,
        defaultValue: DataTypes.NOW,
    },
    updated_at: {
        type: DataTypes.DATE,
        allowNull: true,
    }
}, {
    tableName: 'products',
    timestamps: true,
    underscored: true,
});

module.exports = Product;